pkg/util/provider/machinecontroller/machine.go

```diff
@@ -161,7 +161,7 @@ func (c *controller) reconcileClusterMachine(ctx context.Context, machine *v1alp
 		return retry, err
 	}
 
-	if machine.Status.Node != "" {
+	if machine.Labels[v1alpha1.NodeLabelKey] != "" {
```

Comment:
```suggestion
	if machine.Labels!=nil && machine.Labels[v1alpha1.NodeLabelKey] != "" && machine.Status.CurrentStatus.Phase != "" {
```

This is done to make sure that we are entering this only when the controller has updated the `phase` of the machine which is part of status. if we don't do this then we could enter this block when just the spec is updated and not status.